src/iter.rs

```diff
@@ -1,21 +1,42 @@
-use crate::converter::{Converter, IndexWithConverter};
-
 use crate::character::Character;
+use crate::converter::{Converter, IndexWithConverter};
 use crate::seal;
+use crate::search::Search;
 
-/// A search index that can be searched backwards.
-pub trait BackwardIterableIndex: Sized {
+/// A FM-Index that can search texts backwards and forwards.
+pub trait FMIndex: Sized + seal::Sealed {
     /// A [`Character`] type.
     type T: Character;
 
+    #[doc(hidden)]
+    fn len<L: seal::IsLocal>(&self) -> u64;
+
     #[doc(hidden)]
     fn get_l<L: seal::IsLocal>(&self, i: u64) -> Self::T;
     #[doc(hidden)]
     fn lf_map<L: seal::IsLocal>(&self, i: u64) -> u64;
     #[doc(hidden)]
     fn lf_map2<L: seal::IsLocal>(&self, c: Self::T, i: u64) -> u64;
     #[doc(hidden)]
-    fn len<L: seal::IsLocal>(&self) -> u64;
+    fn get_f<L: seal::IsLocal>(&self, i: u64) -> Self::T;
```

Comment:
Since users cannot utilize the underlying FM-Index structs  (`DefaultFMIndex` or `RLFMIndex`) or implement `FMIndex` for another struct, I believe we can safely remove `seal::IsLocal`.